### ISSUE TITLE
Records won't allow properties to hide names from the base

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#76557]
**Version Used**: https://github.com/dotnet/roslyn/commit/98ea496177bc8e607dbf454abd6b5a5e4678aed2

**Steps to Reproduce**:

```cs
abstract record R1
{
    public sealed record Derived(E E) : R1; // error CS8866 (expected?)
    public enum E { }
}
abstract class R2
{
    public sealed class Derived : R2
    {
        public Derived(E E) => this.E = E;
        public new E E { get; } // notice the `new` modifier
    }
    public enum E { }
}
```

**Expected Behavior**: No errors or at least a warning.

**Actual Behavior**: `error CS8866: Record member 'R1.E' must be a readable instance property or field of type 'R1.E' to match positional parameter 'E'.`
